### DISCUSSION TITLE
React Redux Forms

### DISCUSSION THREAD

**Discussion by jdrgoms** [frontendbr/forum#201] [Discussão]
Boa tarde galera, 
gostaria de saber de vocês que utilizam o react, o que tem usado para forms? Existe alguma opção equivalente ao [****Redux Form****](http://redux-form.com/5.3.1/#/?_k=dkla67)?

**Comment by fdaciuk** [comment]
Eu dei uma olhada no Redux Form, mas acabei criando uma estrutura própria de formulário pra usar com Redux xD

Vi recentemente que modificaram bastante ele, parece estar melhor do que eu vi da primeira vez, mas agora já tenho isso implementado =)

Vou ver se abro o que eu fiz. Seria legal ver outras ideias de implementação =)

**Comment by jdrgoms** [comment]
Realmente @fdaciuk mudou bastante, hoje mesmo o ****Erik Rasmussen**** subiu a [v6.0 rc](http://redux-form.com/6.0.0-rc.1/) .

**Comment by diessica** [comment]
> Existe alguma opção equivalente ao Redux Form?

Você diz  uma opção equivalente sem Redux?  Porque, com Redux, o [`redux-form`](http://redux-form.com/) é a melhor opção.

**Comment by felisio** [comment]
Na verdade @diessica seria uma alternativa ao redux-form pois como estamos fazendo formulários enormes (tipo 50 campos rsrsrsr) o redux-form na versão 5 deu uma "sentada" na performance, perguntei ao Erick e ele me disse que era para migrar-mos para a versão 6... pois bem, antes de continuar essa migração, resolvi da uma pesquisada para ver se tinha outras opções rolando.... valeu..

**Comment by diessica** [comment]
Ah, ter sempre tem. O [`react-redux-form`](https://github.com/davidkpiano/react-redux-form) é uma alternativa, mas ainda assim o Redux Form é mais completo e bem documentado. Eu não o desconsideraria.

Sobre essa questão de má performance em formulários longos, isso pode ser várias coisas. Acontece também sem Redux Forms. A questão é: **Você já averiguou se sua aplicação não está fazendo muitos _re-renders_ desnecessários?** O React permite isso.

A ferramenta [Why Did You Update?](https://github.com/garbles/why-did-you-update) e o [add-on `shallowCompare`](https://facebook.github.io/react/docs/shallow-compare.html) podem ser muito úteis para melhorar a performance da sua aplicação. Na documentação, existem alguns conteúdos interessantes:
- [Performance Tools](https://facebook.github.io/react/docs/perf.html)
- [Advanced Performance](https://facebook.github.io/react/docs/advanced-performance.html)

Caso você já tenha feito tudo isso e o problema seja realmente com o Redux Forms no seu caso, vale a pena considerar o conselho do Erick e considerar uma migração para a versão 6, em vez de correr o risco de trocar para uma biblioteca que possa se mostrar menos completa, estável e performática.

**Comment by jdrgoms** [comment]
Bem interessante. Vamos migrar para a ****versão 6**** mesmo.  Obrigado @diessica & @fdaciuk pelas respostas e dicas. :grinning:

**Comment by felipefialho** [comment]
@diessica sempre mitando nos comentários, haha :)
